src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -394,67 +410,120 @@ int sgn(T val)
-						//{
-						//	break;
-						//}
-
-						// We want the checkered pattern to alternate both vertically and horizontally.
-						// In order to achieve that, we'll toggle visibility of the current pixel on or off
-						// depending on both its x- and its y-position.  If x == CheckerSize, we'll turn visibility off,
-						// but then if y == CheckerSize, we'll turn it back on.
-						// The below is a shorthand for the above intent.
-						bool pixelShouldBeBlank = ((x / CheckerSize) + (y / CheckerSize)) % 2 == 0 ? true : false;
-
-						if (pixelShouldBeBlank)
-						{
-							bgraCheckeredPixelData.Add(0);
-							bgraCheckeredPixelData.Add(0);
-							bgraCheckeredPixelData.Add(0);
-							bgraCheckeredPixelData.Add(0);
-						}
-						else
+						for (int x = 0; x < width; x++)
 						{
-							bgraCheckeredPixelData.Add((byte)(checkerColor.B * checkerColor.A / 255));
-							bgraCheckeredPixelData.Add((byte)(checkerColor.G * checkerColor.A / 255));
-							bgraCheckeredPixelData.Add((byte)(checkerColor.R * checkerColor.A / 255));
-							bgraCheckeredPixelData.Add(checkerColor.A);
+							if (workItem.Status == AsyncStatus.Canceled)
+							{
+								break;
+							}
+
+							// We want the checkered pattern to alternate both vertically and horizontally.
+							// In order to achieve that, we'll toggle visibility of the current pixel on or off
+							// depending on both its x- and its y-position.  If x == CheckerSize, we'll turn visibility off,
+							// but then if y == CheckerSize, we'll turn it back on.
+							// The below is a shorthand for the above intent.
+							bool pixelShouldBeBlank = (x / CheckerSize + y / CheckerSize) % 2 == 0 ? true : false;
+
+							if (pixelShouldBeBlank)
+							{
+								bgraCheckeredPixelData.Add(0);
+								bgraCheckeredPixelData.Add(0);
+								bgraCheckeredPixelData.Add(0);
+								bgraCheckeredPixelData.Add(0);
+							}
+							else
+							{
+								bgraCheckeredPixelData.Add((byte)(checkerColor.B * checkerColor.A / 255));
+								bgraCheckeredPixelData.Add((byte)(checkerColor.G * checkerColor.A / 255));
+								bgraCheckeredPixelData.Add((byte)(checkerColor.R * checkerColor.A / 255));
+								bgraCheckeredPixelData.Add(checkerColor.A);
+							}
 						}
 					}
-				}
-			});
-
-			//if (asyncActionToAssign != null)
-			//{
-			//	asyncActionToAssign.Cancel();
-			//}
+				};
 
-			//asyncActionToAssign = ThreadPool.RunAsync(workItemHandler);
-			//asyncActionToAssign.Completed = new AsyncActionCompletedHandler(
-			//async (IAsyncAction asyncInfo, AsyncStatus asyncStatus) =>
-			//{
-			//	if (asyncStatus != AsyncStatus.Completed)
-			//	{
-			//		return;
-			//	}
-
-			//	asyncActionToAssign = null;
+			if (asyncActionToAssign is { })
+			{
+				asyncActionToAssign.Cancel();
+			}
 
-			// Uno Doc: Assumed normal priority is acceptable
-			await dispatcherHelper.RunAsync(CoreDispatcherPriority.Normal, () =>
+			// Uno Doc: ThreadPool is not implemented, so we work around it
```

Comment:
Oh, ok, you ran into unsupported things here and did Task.

Reply:
Yes, I eventually had to make a workaround like you did, but I tried to keep as much of the original source as possible. This will help if we need to compare it later.

Reply:
This is a grey-area where to me it makes sense to stay with C#/.NET. Code simply doesn't exist to keep it 1-to-1. If you made the decision on the opposite side of that grey area I guess it's fine as long as it still works. It does make sense to keep compatibility as close as possible. I wouldn't have changed it if it was me though.